### PR TITLE
[CDF-25543] 🖍 Convert string to DataType

### PR DESCRIPTION
**Pull request by doctrino**
# Description

**Context**: We have a `csv` reader that naively converts cell values with custom logic. Instead we want to sniff the schema from the file and ensure that all columns have the same data type. 

This PR refactors to reuse the `dtype` conversion logic we have for ViewProperties such that is can be used to convert the values in the csv file once we have the schema.

## Changelog

- [ ] Patch
- [x] Skip

### CODE REVIEW

**Review by Aashutosh-cognite — APPROVED** [review]

File: tests/test_unit/test_cdf_tk/test_utils/test_dtype_conversion.py [hunk 1]
```diff
@@ -484,3 +486,51 @@ def test_asset_centric_failed_conversion(
...
+        pytest.param("true", "boolean", True, id="String 'true' to Boolean"),
+        pytest.param("false", "boolean", False, id="String 'false' to Boolean"),
+        pytest.param('{"key": "value"}', "json", {"key": "value"}, id="Stringified dict to Json"),
+        pytest.param(
+            "2025-07-22T12:34:56Z",
+            "timestamp",
+            datetime(2025, 7, 22, 12, 34, 56, tzinfo=timezone.utc),
+            id="ISO timestamp to Timestamp",
+        ),
+        pytest.param("2025-07-22", "date", date(2025, 7, 22), id="ISO date to Date"),
```
Comment (gaelenm) [inline tests/test_unit/test_cdf_tk/test_utils/test_dtype_conversion.py:505]: Please add a test case for nullable types (or None values).

**Review by gaelenm — APPROVED** [review]
